### ISSUE TITLE
Failed to activate pyzsh due to old zcompdump file

### ISSUE THREAD

**Issue by lavigne958** [dan1994/pyzshcomplete#37]
Hi, I tried your autocomplete tool and for some reasons it failed to remove the '.zcompdump' file in my home dir.

Here is the link to the source of the issue ( I believe) : https://github.com/dan1994/pyzshcomplete/blob/a1975e02497431df1cc7b058226e83104a39169e/scripts/activate_pyzshcomplete#L32

I manually removed the cache files and it worked right away.

To help here the list of the files I had in my home dir:

- .zcompdump-alexander-pc-5.7.1
- .zcompdump-alexander-pc-5.7.1.old
-.zcompdump-alexander-pc-5.8
-.zcompdump.old

What I don't understand is that all the files actually match the patern and should be deleted.

**Comment by lavigne958** [comment]
I think I have an idea, I used 'sudo' to run the script because the script needs to copy files to the system, but when the line I pointed runs it expands the user path to the root home dir (usually `/root/`) 
Therefor the files in my home dir (`/home/alexander`) are not deleted.

**Comment by dan1994** [comment]
Thanks for reporting!
What you say about the home directories makes sense, although from my memory it worked well.
I'll take a look into it.

About the `.zcompdump` files: They are the completion system's cache files that speed up the time to open a shell. I'm not sure why there's more than one, but I delete them on installation to make sure the completion system won't ignore any updated files.

**Comment by dan1994** [comment]
I don't install the package as super user.
If you take a look at the readme you can see the exact commands I use:

```zsh
pip install pyzshcomplete
sudo ~/.local/bin/activate_pyzshcomplete
```

This shouldn't make a difference though. I made a small test where I wrote 2 identical scripts that print `expanduser('~')`. One of the scripts was owned by the user and the other by root. I then invoked both with and without `sudo`. The results are that without `sudo` both print the user's home directory, and with `sudo` both print `/root`.
**TL;DR** When a script is invoked with `sudo`, `expanduser` should always return `/root`. So this is supposedly a bug of mine.

**Comment by dan1994** [comment]
Out of interest, how did you notice the files weren't removed?
Not finding any cache files shouldn't have resulted in an error, since `glob('/root/.zcompdump*)` would have returned an empty list.

**Comment by dan1994** [comment]
Anyways, I changed `activate_pyzshcomplete` to run without `sudo`, and use `subprocess.run` for the single operation that requires root privileges. Should resolve the problem.

**Comment by lavigne958** [comment]
> Out of interest, how did you notice the files weren't removed

Hi, I noticed because I looked at the script `activate_pyzshcomplete` I was curious about what it does.
I don't remember seeing any error message, as you mention it the list is empty so the script just goes on.

> Anyways, I changed `activate_pyzshcomplete` to run without `sudo`, and use `subprocess.run` for the single operation that requires root privileges. Should resolve the problem.

I just looked at the PR, nice. I found this package as well that might interest you: [elevate](https://pypi.org/project/elevate/)

thanks for the fix, I will try it out Monday.

Btw I was looking at your repository because I needed a completion tool for python within zsh, but I mostly use subparsers, I see that you do not handle them right now I would be interested in helping for that matter if you're ok.

**Comment by dan1994** [comment]
> I just looked at the PR, nice. I found this package as well that might interest you: elevate

I saw it too, but decided that I don't want to introduce package dependencies for such a trivial task.

> Btw I was looking at your repository because I needed a completion tool for python within zsh, but I mostly use subparsers, I see that you do not handle them right now I would be interested in helping for that matter if you're ok.

This is true, and I have an open issue for that (#7). 
I would love for some assistance, but I should warn you that this requires a knowledge of `zsh`'s completion system mechanism, and might require some fundamental changes in the way the `zsh` and `python` parts of the program pass information to each other. 
In any case we could talk about it in the issue discussion, or you can mail me directly at dan1994@gmail.com.
